packages/monitor-v2/src/llm-bot/common.ts

```diff
@@ -78,42 +89,47 @@ export abstract class OptimisticOracleClient {
   /**
    * Creates a new instance of the OptimisticOracleClient with the specified requests.
    * Must be implemented by the derived class.
-   * @param provider The provider used for interacting with the blockchain.
    * @param requests The requests to be set on the new instance.
    * @param fetchedBlockRange The block range of the fetched requests.
    * @returns A new instance of OptimisticOracleClient.
    */
   protected abstract createClientInstance(
-    requests: OptimisticOracleRequest[],
+    requests: R[],
     fetchedBlockRange: [number, number]
-  ): OptimisticOracleClient;
+  ): OptimisticOracleClient<R>;
 
   /**
    * Retrieves Optimistic Oracle requests within the specified block range.
    * @param blockRange The block range to fetch requests from.
    * @returns A Promise that resolves to an array of OptimisticOracleRequest objects.
    */
-  protected abstract fetchOracleRequests(blockRange: [number, number]): Promise<OptimisticOracleRequest[]>;
+  protected abstract fetchOracleRequests(blockRange: [number, number]): Promise<R[]>;
 
   /**
    * Updates the OptimisticOracleClient instance by fetching new Oracle requests within the specified block range. Returns a new instance.
-   * @param blockRange The block range to fetch new requests from.
+   * @param blockRange (Optional) The block range to fetch new requests from.
    * @param existingRequests (Optional) The list of existing requests to merge with the new requests.
    * @returns A Promise that resolves to a new OptimisticOracleClient instance with updated requests.
    */
-  async updateWithBlockRange(
-    existingRequests: OptimisticOracleRequest[] = [],
-    blockRange: [number, number]
-  ): Promise<OptimisticOracleClient> {
-    const newRequests = await this.fetchOracleRequests(blockRange);
-    return this.createClientInstance([...existingRequests, ...newRequests], blockRange);
+  async updateWithBlockRange(blockRange?: [number, number], existingRequests: R[] = []): Promise<this> {
+    let range: [number, number];
+    if (blockRange) {
+      range = blockRange;
+    } else {
+      const latestBlock = await this.provider.getBlockNumber();
+      range = [this.fetchedBlockRange[1] + 1, latestBlock];
+    }
+    if (range[0] > range[1]) throw new Error("Invalid block range");
+    if (range[0] == range[1]) return this.createClientInstance(existingRequests, range) as this;
```

Comment:
If there were no new blocks mined since the last update (latestBlock == this.fetchedBlockRange[1]) the `(range[0] > range[1])` could be true due to `range = [this.fetchedBlockRange[1] + 1, latestBlock]` above. So, in this particular scenario we don't want to throw, but return the existing range requests.

Also, the second check `(range[0] == range[1])` being true does not mean that there are no new requests. In case there was 1 block mined since the previous update the range could equal `[this.fetchedBlockRange[1] + 1, this.fetchedBlockRange[1] + 1]` and we would want to fetch oracle requests for this 1 block range.

Reply:
Thanks @Reinis-FRP,
Fixed in latest commit.